### PR TITLE
chore: refactor notification center

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Refactored to align with latest component designs. 
- introduce component schema
- fixed broken storybook
- introduce new alignment features: `center` position (top-center, bottom-center)
- introduced configurable margins and size through component options

closes  [HRZ-2374](https://spryker.atlassian.net/browse/HRZ-2374) [HRZ-2375](https://spryker.atlassian.net/browse/HRZ-2375)

[HRZ-2374]: https://spryker.atlassian.net/browse/HRZ-2374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HRZ-2375]: https://spryker.atlassian.net/browse/HRZ-2375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/base/ui/overlays/notification-center/src/notification-center.component.ts [hunk 1]
```diff
@@ -23,7 +23,22 @@ export const defaultStrategy: NotificationStrategy = {
 export class NotificationCenterComponent extends LitElement {
   static styles = [notificationCenterBaseStyles];
 
-  @property({ reflect: true }) position?: Positions;
+  @property({ reflect: true }) position?: NotificationPosition;
+
+  @property() set maxWidth(value: string) {
```
Comment (Sinucid) [inline libs/base/ui/overlays/notification-center/src/notification-center.component.ts]: I didn't get, why do we need these properties? Why not just set these css properies in css files or even is style prop?

I would suggest to define default values for them in oryx-notification-center styles and if need to override them just do this by styles

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/base/ui/overlays/notification-center/src/notification-center.component.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [reply libs/base/ui/overlays/notification-center/src/notification-center.component.ts]: i started like that and have 2nd thoughts. Will give it another go, thx

Comment (Sinucid) [reply libs/base/ui/overlays/notification-center/src/notification-center.component.ts]: Now I got. But still...
```

protected applyStyles(): string | void {
   const {maxWidth, marginBlock, marginInline} = this.componentOptions ?? {};
   return `${maxWidth ? `--oryx-notification-max-width: ${maxWidth};` : ''}
   ${marginBlock ? `--oryx-notification-margin-block: ${marginBlock};` : ''}
   ${marginInline ? `--oryx-notification-margin-inline: ${marginInline};` : ''}
   `.trim() || undefined;
}

<oryx-notification-center
      .position=${this.componentOptions?.position}
      styles=${ifDefined(this.applyStyles())}
      ${ref(this.centerRef)}
    ></oryx-notification-center>`
```

File: libs/domain/site/notification-center/src/notification-center.component.ts [hunk 2]
```diff
@@ -35,10 +36,22 @@ export class SiteNotificationCenterComponent extends ContentMixin<SiteNotificati
   protected override render(): TemplateResult {
     return html`<oryx-notification-center
       .position=${this.componentOptions?.position}
-      .marginBlock=${this.componentOptions?.marginBlock}
-      .marginInline=${this.componentOptions?.marginInline}
-      .maxWidth=${this.componentOptions?.maxWidth}
+      style=${ifDefined(this.getStyle())}
```
Comment (Sinucid) [inline libs/domain/site/notification-center/src/notification-center.component.ts:44]: getStyle returns string every time, ifDefined is useless

**Review by Sinucid — CHANGES_REQUESTED** [review]

File: libs/domain/site/notification-center/src/notification-center.component.spec.ts [hunk 3]
```diff
@@ -96,22 +96,66 @@ describe('SiteNotificationCenterComponent', () => {
...
+      ) as NotificationCenterComponent;
+      expect(
+        notificationCenter.style.getPropertyValue(
+          '--oryx-notification-margin-block'
+        )
+      ).toBe('10vh');
+    });
+  });
+
+  describe.only('when marginInline option is provided', () => {
```
Comment (Sinucid) [inline libs/domain/site/notification-center/src/notification-center.component.spec.ts]: only

File: libs/domain/site/notification-center/src/notification-center.component.spec.ts [hunk 4]
```diff
@@ -96,22 +96,66 @@ describe('SiteNotificationCenterComponent', () => {
...
+          '--oryx-notification-max-width'
+        )
+      ).toBe('50%');
+    });
+  });
 
-      const open = vi.spyOn(notificationCenter, 'open');
-      notificationTrigger$.next(mockNotification);
-      expect(open).toHaveBeenCalledWith(mockNotification);
+  describe.only('when marginBlock option is provided', () => {
```
Comment (Sinucid) [inline libs/domain/site/notification-center/src/notification-center.component.spec.ts]: Only

File: libs/domain/site/notification-center/src/notification-center.component.ts (hunk 2 above)
Comment (Sinucid) [reply libs/domain/site/notification-center/src/notification-center.component.ts:44]: What about this?

**Review by Sinucid — CHANGES_REQUESTED** [review]

**Review by Sinucid — DISMISSED** [review]

File: libs/base/ui/graphical/chip/src/chip.component.spec.ts [hunk 5]
```diff
@@ -17,25 +18,29 @@ describe('ChipComponent', () => {
...
-          });
-
-          it('passes the a11y audit', async () => {
-            await expect(element).shadowDom.to.be.accessible(a11yConfig);
-          });
-
-          it('should reflect the type attribute', () => {
-            expect(element?.getAttribute('appearance')).toBe(appearance);
-          });
+    Object.values([
```
Comment (supproduction) [inline libs/base/ui/graphical/chip/src/chip.component.spec.ts]:
```suggestion
    [
```

File: libs/base/ui/overlays/notification/src/stories/Static/notification-states.stories.ts [hunk 6]
```diff
@@ -82,27 +83,48 @@ const Template: Story = ({ backgroundColor }): TemplateResult => {
...
           <section>
             <h3>Closable (${scheme})</h3>
-            ${Object.keys(Types).map((type) =>
-              getNotification(type, scheme, true)
-            )}
+            ${[
+              AlertType.Info,
+              AlertType.Success,
+              AlertType.Warning,
+              AlertType.Error,
```
Comment (supproduction) [inline libs/base/ui/overlays/notification/src/stories/Static/notification-states.stories.ts]: Can we move this list into local variable and reuse it?

**Review by supproduction — CHANGES_REQUESTED** [review]

**Review by supproduction — APPROVED** [review]

**Pull request merged**